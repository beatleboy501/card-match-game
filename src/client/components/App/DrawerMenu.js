import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import PlayArrowIcon from '@material-ui/icons/PlayArrow';
import LooksTwoIcon from '@material-ui/icons/LooksTwo';
import LooksThreeIcon from '@material-ui/icons/Looks3';
import CloseIcon from '@material-ui/icons/Close';
import RefreshIcon from '@material-ui/icons/Refresh';
import KeyboardReturnIcon from '@material-ui/icons/KeyboardReturn';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import IconButton from '@material-ui/core/IconButton';
import CardMatchCounts from '../../constants/CardMatchCounts';

const listItemsPropTypes = {
  onCloseDrawer: PropTypes.func.isRequired,
  isGame: PropTypes.bool.isRequired,
  toggleSpotifyWidget: PropTypes.func.isRequired,
  toggleCardMatchCount: PropTypes.func.isRequired,
  onResetGame: PropTypes.func.isRequired,
  onExitGame: PropTypes.func.isRequired,
};

const generateListItem = (onClick, primary, icon) => (
  <ListItem
    button
    onClick={onClick}
  >
    <ListItemIcon>
      {icon}
    </ListItemIcon>
    <ListItemText primary={primary} />
  </ListItem>
);

const listItems = (props) => {
  const {
    onCloseDrawer, isGame, toggleSpotifyWidget, toggleCardMatchCount, onResetGame, onExitGame,
  } = props;
  return (
    <div>
      {generateListItem(() => {
        toggleSpotifyWidget();
        onCloseDrawer();
      }, 'Spotify Widget', <PlayArrowIcon />)}
      {!isGame && generateListItem(() => {
        toggleCardMatchCount(CardMatchCounts.TWO_CARD_MATCH);
        onCloseDrawer();
      }, '2 Card Match', <LooksTwoIcon />)}
      {!isGame && generateListItem(() => {
        toggleCardMatchCount(CardMatchCounts.THREE_CARD_MATCH);
        onCloseDrawer();
      }, '3 Card Match', <LooksThreeIcon />)}
      {isGame && generateListItem(() => {
        onCloseDrawer();
        onResetGame();
      }, 'Reset Game', <RefreshIcon />)}
      {isGame && generateListItem(() => {
        onCloseDrawer();
        onExitGame();
      }, 'Main Menu', <KeyboardReturnIcon />)}
    </div>
  );
};

listItems.propTypes = listItemsPropTypes;

const propTypes = {
  onCloseDrawer: PropTypes.func.isRequired,
  isDrawerDisplayed: PropTypes.bool.isRequired,
};

const DrawerMenu = (props) => {
  const {
    onCloseDrawer, isDrawerDisplayed,
  } = props;
  return (
    <Drawer anchor="left" open={isDrawerDisplayed}>
      <div>
        <IconButton onClick={onCloseDrawer}>
          <CloseIcon />
        </IconButton>
      </div>
      <Divider />
      <List>{listItems(props)}</List>
      <Divider />
    </Drawer>
  );
};

DrawerMenu.propTypes = propTypes;

export default DrawerMenu;
