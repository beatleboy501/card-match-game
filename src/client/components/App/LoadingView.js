import React from 'react';

const LoadingView = () => (<div>Loading...</div>);

export default LoadingView;
