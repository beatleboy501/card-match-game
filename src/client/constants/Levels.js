export default Object.freeze({
  EASY: 'easy',
  HARD: 'hard',
});
