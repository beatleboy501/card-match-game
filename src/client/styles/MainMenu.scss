$hey-look-at-me-im-a-sass-variable: #333;

.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.placeholder {
  font-style: italic;
}

.mainPanel {
  align-items: center;
  text-align: center;
  width: 100vw;
}

.logo {
  width: 50px;
  height: 50px;
}
