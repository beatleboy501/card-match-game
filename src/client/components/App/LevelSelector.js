import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import styles from '../../styles/LevelSelector.scss';

const propTypes = {
  fetchCards: PropTypes.func.isRequired,
  levels: PropTypes.arrayOf(PropTypes.object),
};

const defaultProps = {
  levels: [],
};

const LevelSelector = (props) => {
  const { fetchCards, levels } = props;
  return (
    <div className="level-selector">
      {levels.length ? (
        <Typography className={styles.heading} variant="title" color="inherit">
          Choose a Level
        </Typography>
      ) : <div />}
      <div className={styles.buttonSet}>
        {levels.map(level => (
          <Button
            key={level.difficulty}
            className={styles.button}
            onClick={() => fetchCards(level.difficulty)}
            type="button"
          >
            {level.difficulty}
          </Button>
        ))}
      </div>
    </div>
  );
};

LevelSelector.propTypes = propTypes;
LevelSelector.defaultProps = defaultProps;

export default LevelSelector;
