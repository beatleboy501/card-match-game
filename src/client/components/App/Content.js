import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import MemoryGame from '../MemoryGame/index';
import LevelSelector from './LevelSelector';
import styles from '../../styles/MiddleContent.scss';

const propTypes = {
  currentCards: PropTypes.arrayOf(PropTypes.object),
  levels: PropTypes.arrayOf(PropTypes.object),
  fetchCards: PropTypes.func.isRequired,
  onGameStart: PropTypes.func.isRequired,
  onGameComplete: PropTypes.func.isRequired,
  parentRef: PropTypes.func.isRequired,
  cardMatchCount: PropTypes.number.isRequired,
};

const defaultProps = {
  currentCards: null,
  levels: [],
};

const Content = (props) => {
  const {
    currentCards, levels, fetchCards, onGameStart,
    onGameComplete, parentRef, cardMatchCount,
  } = props;
  return (
    <div className={styles.mainPanel}>
      <Paper className={styles.paper} elevation={10}>
        {currentCards ? (
          <MemoryGame
            ref={el => parentRef(el)}
            cardMatchCount={cardMatchCount}
            cards={currentCards}
            onGameStart={onGameStart}
            onGameComplete={onGameComplete}
          />
        ) : (
          <LevelSelector levels={levels} fetchCards={fetchCards} />
        )}
      </Paper>
    </div>
  );
};


Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;
