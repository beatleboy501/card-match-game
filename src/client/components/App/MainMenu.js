import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  closeDrawer, openDrawer, resetGame, setCurrentCards,
  setLevels, toggleSpotifyWidget, toggleCardMatchCount,
  persistRehydrate,
} from '../../redux/actions/mainMenuActions';
import { resetMemoryGame } from '../../redux/actions/memoryGameActions';
import logo from '../../../public/timesLogo.png';
import { gameEndpoint, offlineLevels } from '../../utils';
import TopNav from './TopNav';
import Content from './Content';
import styles from '../../styles/MainMenu.scss';
import DrawerMenu from './DrawerMenu';
import SpotifyWidget from './SpotifyWidget';
import CardMatchCounts from '../../constants/CardMatchCounts';

const propTypes = {
  currentCards: PropTypes.arrayOf(PropTypes.object),
  isDrawerDisplayed: PropTypes.bool.isRequired,
  isSpotifyWidgetDisplayed: PropTypes.bool.isRequired,
  cardMatchCount: PropTypes.number,
  levels: PropTypes.arrayOf(PropTypes.object),
  dispatchCloseDrawer: PropTypes.func.isRequired,
  dispatchOpenDrawer: PropTypes.func.isRequired,
  dispatchResetGame: PropTypes.func.isRequired,
  dispatchResetMemoryGame: PropTypes.func.isRequired,
  dispatchSetLevels: PropTypes.func.isRequired,
  dispatchSetCurrentCards: PropTypes.func.isRequired,
  dispatchToggleSpotifyWidget: PropTypes.func.isRequired,
  dispatchToggleCardMatchCount: PropTypes.func.isRequired,
};

const defaultProps = {
  currentCards: null,
  levels: [],
  cardMatchCount: CardMatchCounts.TWO_CARD_MATCH,
};

export class MainMenu extends React.Component {
  constructor(props) {
    super(props);
    this.fetchCards = this.fetchCards.bind(this);
    this.reset = this.reset.bind(this);
    this.displayLevelSelector = this.displayLevelSelector.bind(this);
    this.handleGameStart = this.handleGameStart.bind(this);
    this.handleGameComplete = this.handleGameComplete.bind(this);
    this.state = {};
  }

  componentDidMount() { // could also move this to middleware
    const { levels, dispatchSetLevels } = this.props;
    if (levels.length > 0) return;
    fetch(gameEndpoint)
      .then(response => response.json())
      .then((result) => {
        const resultLevels = result.levels;
        dispatchSetLevels(resultLevels);
      })
      .catch((error) => {
        console.error(error);
        dispatchSetLevels(offlineLevels);
      });
  }

  displayLevelSelector() {
    this.reset();
    const { dispatchSetCurrentCards } = this.props;
    dispatchSetCurrentCards(null);
  }

  fetchCards(level) {
    const { dispatchSetCurrentCards, levels, cardMatchCount } = this.props;
    const levelCards = levels.length ? levels.find(l => l.difficulty === level) : null;
    if (!levelCards) {
      alert(`Invalid level: ${level}`);
    } else {
      const cards = cardMatchCount === CardMatchCounts.TWO_CARD_MATCH
        ? levelCards.twoCardMatch
        : levelCards.threeCardMatch;
      dispatchSetCurrentCards(cards);
    }
  }

  handleGameStart() {
    this.timer.start();
  }

  handleGameComplete(matchesCount) {
    this.timer.stop();
    const time = this.timer.getTime();
    alert(`${matchesCount} matches in ${time.minutes}:${time.seconds}:${time.hundredths}`);
  }

  reset() {
    const { dispatchResetGame, dispatchResetMemoryGame } = this.props;
    this.timer.reset();
    this.setState({});
    dispatchResetMemoryGame();
    dispatchResetGame();
  }

  render() {
    const {
      cardMatchCount,
      currentCards,
      dispatchCloseDrawer,
      dispatchOpenDrawer,
      dispatchToggleCardMatchCount,
      dispatchToggleSpotifyWidget,
      isDrawerDisplayed,
      isSpotifyWidgetDisplayed,
      levels,
    } = this.props;
    return (
      <div className={styles.container}>
        <TopNav
          currentCards={currentCards}
          logo={logo}
          timerRef={(el) => { this.timer = el && el.getWrappedInstance(); }}
          onBurgerClick={dispatchOpenDrawer}
        />
        <DrawerMenu
          onCloseDrawer={dispatchCloseDrawer}
          isGame={!!currentCards}
          isDrawerDisplayed={isDrawerDisplayed}
          toggleSpotifyWidget={dispatchToggleSpotifyWidget}
          toggleCardMatchCount={dispatchToggleCardMatchCount}
          onResetGame={this.reset}
          onExitGame={this.displayLevelSelector}
        />
        <Content
          currentCards={currentCards}
          levels={levels}
          fetchCards={this.fetchCards}
          parentRef={(el) => { this.memoryGame = el; }}
          onGameStart={this.handleGameStart}
          onGameComplete={this.handleGameComplete}
          cardMatchCount={cardMatchCount}
        />
        {isSpotifyWidgetDisplayed && (
          <SpotifyWidget />
        )}
      </div>
    );
  }
}

MainMenu.propTypes = propTypes;
MainMenu.defaultProps = defaultProps;

function mapStateToProps(state) {
  return { ...state.mainMenu };
}

const mapDispatchToProps = dispatch => ({
  dispatchResetGame: () => {
    dispatch(resetGame());
  },
  dispatchResetMemoryGame: () => {
    dispatch(resetMemoryGame());
  },
  dispatchSetCurrentCards: (currentCards) => {
    dispatch(setCurrentCards(currentCards));
  },
  dispatchSetLevels: (levels) => {
    dispatch(setLevels(levels));
  },
  dispatchOpenDrawer: () => {
    dispatch(openDrawer());
  },
  dispatchCloseDrawer: () => {
    dispatch(closeDrawer());
  },
  dispatchToggleSpotifyWidget: () => {
    dispatch(toggleSpotifyWidget());
  },
  dispatchToggleCardMatchCount: (cardMatchCount) => {
    dispatch(toggleCardMatchCount(cardMatchCount));
  },
  dispatchPersistRehydrate: (payload) => {
    dispatch(persistRehydrate(payload));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(MainMenu);
