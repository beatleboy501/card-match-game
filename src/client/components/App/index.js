import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/lib/integration/react';
import MainMenu from './MainMenu';
import { persistor, store } from '../../redux/store/configureStore';
import LoadingView from './LoadingView';

const App = () => (
  <Provider store={store}>
    <PersistGate loading={<LoadingView />} persistor={persistor}>
      <MainMenu />
    </PersistGate>
  </Provider>
);

export default App;
