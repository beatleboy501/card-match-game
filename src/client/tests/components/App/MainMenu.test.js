import React from 'react';
import { Provider } from 'react-redux';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { MainMenu } from '../../../components/App/MainMenu';
import Levels from '../../../constants/Levels';
import { easyCards, levels, resetProps } from '../../testUtils';
import 'isomorphic-fetch';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const mockCloseDrawer = jest.fn(() => {});
const mockOpenDrawer = jest.fn(() => {});
const mockResetGame = jest.fn(() => {});
const mockResetMemoryGame = jest.fn(() => {});
const mockSetLevels = jest.fn(() => {});
const mockSetCurrentCards = jest.fn(() => {});
const mockToggleSpotifyWidget = jest.fn(() => {});
const mockToggleCardMatchCount = jest.fn(() => {});

let spy;
let store;
let component;
let defaultProps;
global.alert = jest.fn(message => message);

beforeEach(() => {
  defaultProps = {
    currentCards: null,
    levels: [],
    isDrawerDisplayed: false,
    isSpotifyWidgetDisplayed: false,
    dispatchToggleSpotifyWidget: mockToggleSpotifyWidget,
    dispatchToggleCardMatchCount: mockToggleCardMatchCount,
    dispatchCloseDrawer: mockCloseDrawer,
    dispatchOpenDrawer: mockOpenDrawer,
    dispatchResetGame: mockResetGame,
    dispatchResetMemoryGame: mockResetMemoryGame,
    dispatchSetLevels: mockSetLevels,
    dispatchSetCurrentCards: mockSetCurrentCards,
  };
  store = mockStore(defaultProps);
});

afterEach(() => {
  defaultProps = resetProps(defaultProps);
});

it('Renders without crashing', () => {
  component = shallow(<MainMenu {...defaultProps} />).shallow();
  expect(component).toMatchSnapshot();
});

it('Fetches levels after mount', () => {
  spy = jest.spyOn(MainMenu.prototype, 'componentDidMount');
  component = mount(<MainMenu store={store} {...defaultProps} />);
  component.instance().componentDidMount();
  expect(spy).toHaveBeenCalled();
  spy.mockClear();
});

it('Sets cards when fetchCards is called with a valid level', () => {
  spy = jest.spyOn(MainMenu.prototype, 'fetchCards');
  component = mount(
    <MainMenu
      store={store}
      {...defaultProps}
      levels={levels}
    />,
  );
  component.instance().fetchCards(Levels.EASY);
  expect(spy).toHaveBeenCalledWith(Levels.EASY);
  expect(mockSetCurrentCards).toHaveBeenCalled();
  spy.mockClear();
});

it('Returns MemoryGame cards when currentCards has been set', () => {
  component = mount(
    <Provider store={store}>
      <MainMenu
        {...defaultProps}
        levels={levels}
        currentCards={easyCards}
      />
    </Provider>,
  );
  expect(component.html()).toContain('div class="Flipper"');
});

it('Returns LevelSelector when fetchCards is called with an invalid level', () => {
  component = mount(
    <MainMenu
      {...defaultProps}
      store={store}
      levels={levels}
    />,
  );
  component.find(MainMenu).instance().fetchCards('INVALID');
  expect(defaultProps.dispatchSetCurrentCards).not.toHaveBeenCalled();
  expect(component.render().find('div.Flipper').length).toBeLessThan(1);
  expect(component.render().find('div.level-selector').length).toBeGreaterThan(0);
});

it('Returns LevelSelector when no level selected', () => {
  component = mount(
    <MainMenu
      {...defaultProps}
      store={store}
      levels={levels}
    />,
  );
  expect(defaultProps.dispatchSetCurrentCards).not.toHaveBeenCalled();
  expect(component.render().find('div.Flipper').length).toBeLessThan(1);
  expect(component.render().find('div.level-selector').length).toBeGreaterThan(0);
});
