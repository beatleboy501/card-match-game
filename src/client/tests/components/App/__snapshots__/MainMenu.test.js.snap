// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="container"
  >
    <TopNav
      currentCards={null}
      logo="file-stub"
      onBurgerClick={[MockFunction]}
      timerRef={[Function]}
    />
    <DrawerMenu
      isDrawerDisplayed={false}
      isGame={false}
      onCloseDrawer={[MockFunction]}
      onExitGame={[Function]}
      onResetGame={[Function]}
      toggleCardMatchCount={[MockFunction]}
      toggleSpotifyWidget={[MockFunction]}
    />
    <Content
      cardMatchCount={2}
      currentCards={null}
      fetchCards={[Function]}
      levels={Array []}
      onGameComplete={[Function]}
      onGameStart={[Function]}
      parentRef={[Function]}
    />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TopNav
          currentCards={null}
          logo="file-stub"
          onBurgerClick={[MockFunction]}
          timerRef={[Function]}
        />,
        <DrawerMenu
          isDrawerDisplayed={false}
          isGame={false}
          onCloseDrawer={[MockFunction]}
          onExitGame={[Function]}
          onResetGame={[Function]}
          toggleCardMatchCount={[MockFunction]}
          toggleSpotifyWidget={[MockFunction]}
        />,
        <Content
          cardMatchCount={2}
          currentCards={null}
          fetchCards={[Function]}
          levels={Array []}
          onGameComplete={[Function]}
          onGameStart={[Function]}
          parentRef={[Function]}
        />,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentCards": null,
          "logo": "file-stub",
          "onBurgerClick": [MockFunction],
          "timerRef": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isDrawerDisplayed": false,
          "isGame": false,
          "onCloseDrawer": [MockFunction],
          "onExitGame": [Function],
          "onResetGame": [Function],
          "toggleCardMatchCount": [MockFunction],
          "toggleSpotifyWidget": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cardMatchCount": 2,
          "currentCards": null,
          "fetchCards": [Function],
          "levels": Array [],
          "onGameComplete": [Function],
          "onGameStart": [Function],
          "parentRef": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TopNav
            currentCards={null}
            logo="file-stub"
            onBurgerClick={[MockFunction]}
            timerRef={[Function]}
          />,
          <DrawerMenu
            isDrawerDisplayed={false}
            isGame={false}
            onCloseDrawer={[MockFunction]}
            onExitGame={[Function]}
            onResetGame={[Function]}
            toggleCardMatchCount={[MockFunction]}
            toggleSpotifyWidget={[MockFunction]}
          />,
          <Content
            cardMatchCount={2}
            currentCards={null}
            fetchCards={[Function]}
            levels={Array []}
            onGameComplete={[Function]}
            onGameStart={[Function]}
            parentRef={[Function]}
          />,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentCards": null,
            "logo": "file-stub",
            "onBurgerClick": [MockFunction],
            "timerRef": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isDrawerDisplayed": false,
            "isGame": false,
            "onCloseDrawer": [MockFunction],
            "onExitGame": [Function],
            "onResetGame": [Function],
            "toggleCardMatchCount": [MockFunction],
            "toggleSpotifyWidget": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cardMatchCount": 2,
            "currentCards": null,
            "fetchCards": [Function],
            "levels": Array [],
            "onGameComplete": [Function],
            "onGameStart": [Function],
            "parentRef": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
